test/fees/CompoundBorrowFeeCalculator.t.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Test} from 'forge-std/Test.sol';
+import {IERC20} from 'forge-std/interfaces/IERC20.sol';
+import {Router} from 'src/Router.sol';
+import {FeeCalculatorBase} from 'src/fees/FeeCalculatorBase.sol';
+import {CompoundBorrowFeeCalculator} from 'src/fees/CompoundBorrowFeeCalculator.sol';
+import {IParam} from 'src/interfaces/IParam.sol';
+import {IAgent} from 'src/interfaces/IAgent.sol';
+
+interface IComet {
+    struct AssetInfo {
+        uint8 offset;
+        address asset;
+        address priceFeed;
+        uint64 scale;
+        uint64 borrowCollateralFactor;
+        uint64 liquidateCollateralFactor;
+        uint64 liquidationFactor;
+        uint128 supplyCap;
+    }
+
+    function getAssetInfoByAddress(address asset) external view returns (AssetInfo memory);
+
+    function baseToken() external view returns (address);
+
+    function baseBorrowMin() external view returns (uint);
+
+    function supply(address asset, uint amount) external;
+
+    function allow(address manager, bool isAllowed) external;
+}
+
+contract CompoundBorrowFeeCalculatorTest is Test {
+    event FeeCharged(address indexed token, uint256 amount, bytes32 metadata);
+
+    address public constant NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant C_USDCV3 = 0xc3d688B66703497DAA19211EEdff47f25384cdc3;
+    address public constant DUMMY_TO_ADDRESS = address(0);
+    bytes32 public constant META_DATA = bytes32(bytes('compound-v3:borrow'));
+    bytes4 public constant COMPOUND_BORROW_SELECTOR =
+        bytes4(keccak256(bytes('withdrawFrom(address,address,address,uint256)')));
```

Comment:
Curious how to distinguish between the user wants Withdraw and Borrow?

Reply:
It's hard to distinguish user want to Withdraw or Borrow. The discussion conclusion is [here](https://app.asana.com/0/0/1204244120891484/1204519489763734/f)

Reply:
I see.